### PR TITLE
Updates sibling namefittings

### PR DESCRIPTION
**Pull request by domna**
The sibling namefitting was still unreliable. This PR should fix it.

### CODE REVIEW

File: src/pynxtools/dataconverter/nexus_tree.py [hunk 1]
```diff
@@ -245,6 +245,9 @@ def search_child_with_name(
                 return self.add_inherited_node(name)
         return None
 
+    def get_children_for(self, xml_elem: ET._Element) -> Optional[ET._Element]:
+        return next((x for x in self.children if x.inheritance[0] == xml_elem), None)
```
Comment (lukaspie) [inline src/pynxtools/dataconverter/nexus_tree.py:249]: I guess this could benefit from a docstring.

**Review by lukaspie — APPROVED** [review]
LGTM, maybe some docs might be helpful for future maintenance.

File: src/pynxtools/dataconverter/nexus_tree.py (hunk 1 above)
Comment (domna) [reply src/pynxtools/dataconverter/nexus_tree.py:249]: Ah yes, I forgot to write it. Thanks I'll update it

Comment (domna) [reply src/pynxtools/dataconverter/nexus_tree.py:260]: Added

**Pull request merged**